numpy/core/src/multiarray/number.c

```diff
@@ -565,7 +565,7 @@ array_negative(PyArrayObject *m1)
 static PyObject *
 array_absolute(PyArrayObject *m1)
 {
-    if (can_elide_temp_unary(m1)) {
+    if (can_elide_temp_unary(m1) && !PyArray_ISCOMPLEX(m1)) {
```

Comment:
Isn't this also going to fail for any user type that overrides `abs` too? #4730 comes to mind, and problems with a `quaternion` type.